### PR TITLE
Update M365DSCDRGUtil.psm1 to support Multiple Clouds

### PR DESCRIPTION
**Pull request by isaacl1107**
Intune endpoints were all hardcoded for graph.microsoft.com -- rendering the intune pull useless for GCC-H and DoD Cloud -- updated to pull the Graph Base URL from the existing  $Global:MSCloudLoginConnectionProfile.Intune.GraphBaseUrl variable

### CODE REVIEW

**Comment by andikrueger** [comment]
Good find. Could you please add an entry within the changelog?

**Comment by isaacl1107** [comment]
@microsoft-github-policy-service agree

**Comment by isaacl1107** [comment]
@microsoft-github-policy-service agree

File: Modules/Microsoft365DSC/Modules/M365DSCDRGUtil.psm1 [hunk 1]
```diff
@@ -400,7 +400,7 @@ function Get-M365DSCDRGComplexTypeToString
                 $currentValue = $ComplexObject[$key]
                 if ($currentValue.GetType().Name -eq 'String')
                 {
-                    $currentValue = $ComplexObject[$key].Replace("'", "''").Replace("�", "''")
+                    $currentValue = $ComplexObject[$key].Replace("'", "''").Replace(" ", "''")
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/Modules/M365DSCDRGUtil.psm1:403]: Why this change?

**Review by NikCharlebois — CHANGES_REQUESTED** [review]

File: Modules/Microsoft365DSC/Modules/M365DSCDRGUtil.psm1 (hunk 1 above)
Comment (isaacl1107) [reply Modules/Microsoft365DSC/Modules/M365DSCDRGUtil.psm1:403]: I didn't explicitly modify this line -- this might have been an artifact from VSCode -- that can be ignored.

Comment (NikCharlebois) [reply Modules/Microsoft365DSC/Modules/M365DSCDRGUtil.psm1]: Any chance you can remove the change? I can't merge otherwise, you control the code for the PR. Thanks

Comment (isaacl1107) [reply Modules/Microsoft365DSC/Modules/M365DSCDRGUtil.psm1]: I've removed the change in the latest commit

**Review by NikCharlebois — APPROVED** [review]
